### PR TITLE
Refactor preview-backend

### PR DESCRIPTION
**Pull request by avaer**

### CODE REVIEW

File: routes/preview.js [hunk 1]
```diff
@@ -1,228 +1,236 @@
...
-  return p;
-};
-
-const _warn = err => {
-  console.warn('uncaught: ' + err.stack);
-};
-process.on('uncaughtException', _warn);
-process.on('unhandledRejection', _warn);
+const idRegex = /^\/([0-9]+)/;
+const queryA = /^\/\[([^\]]+)\.([^\].]+)]\/([^.]+)\.(.+)$/;
```
Comment (avaer) [inline routes/preview.js:26]: Could use a better name.

File: README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ Node server hosted on AWS, mainly used for REST endpoints.
 
 You can get these credentials from Avaer.
 
-3. Create a `cert/` folder. Create 2 new files inside: `fullchain.pem` and `privkey.pem`. Ask Avaer for the certificates.
+3. Run `npm run script:generateCerts`
```
Comment (avaer) [inline README.md:20]: I suspect the browser will not like this; the certs need to not be self-signed.

Comment (soulofmischief) [reply README.md:20]: The browser is fine with it, I run Chrome with --ignore-certificate-errors during development.
Also seems not to be the root of the access denied issue. So for now it *might* be OK to use locally signed certs (and much more secure) but this warrants further investigation.

Comment (avaer) [reply README.md:20]: > I run Chrome with --ignore-certificate-errors during development.

Don't do that, it is a very wild variance from production, and IMO an easy way to introduce bugs that will break only when you deploy.

Instead we can just copy over the certs.

File: routes/preview.js (hunk 1 above)
Comment (avaer) [reply routes/preview.js:26]: This query is for the `[https://]` URL case.

File: routes/preview.js [hunk 3]
```diff
@@ -1,228 +1,236 @@
...
+const queryB = /^\/([^.]+)\.([^\/]+)\/([^.]+)\.(.+)$/;
```
Comment (avaer) [inline routes/preview.js:27]: This query is for the hash/ext/type case.

File: routes/preview.js (hunk 1 above)
Comment (avaer) [reply routes/preview.js:26]: In general these regexes should not be constantized like this.

They are highly tied to the URL lookup functions they perform and are not used anywhere else.

**Comment by avaer** [comment]
https://github.com/webaverse/preview-backend/commit/35c6e8ae9991591d31d4cefc028e74378868e400

**Pull request closed without merging**